service/frontend/service.go

```diff
@@ -345,11 +348,15 @@ func (s *Service) Start() {
 	s.operatorHandler.Start()
 	s.handler.Start()
 
-	listener := s.grpcListener
-	logger.Info("Starting to serve on frontend listener")
-	if err := s.server.Serve(listener); err != nil {
-		logger.Fatal("Failed to serve on frontend listener", tag.Error(err))
-	}
+	go func() {
```

Comment:
This function is supposed to block and this'll make it not block anymore.. I think this will have effects on shutdown: effectively it'll make the process exit before properly shutting things down. I think... it's confusing

See https://github.com/temporalio/temporal/blob/master/service/frontend/fx.go#L602-L624 and https://github.com/temporalio/temporal/blob/master/temporal/server_impl.go#L106-L124 and https://github.com/temporalio/temporal/blob/master/temporal/fx.go#L312-L326

Unless you want to refactor the whole stopChan thing (not a bad idea but probably out of scope) I think you should add a channel or waitgroup here to block

Reply:
@dnr : thank you for pointing this out. I discussed with @MichaelSnowden , and thought we could address this via doing the membership start in a goroutine vs the Serve call, so I've swapped those two in the latest change, PTAL.